src/main/java/com/shaft/cucumber/GuiActionSteps.java

```diff
@@ -0,0 +1,115 @@
+package com.shaft.cucumber;
+
+import com.shaft.gui.browser.BrowserActions;
+import com.shaft.gui.browser.BrowserFactory;
+import com.shaft.gui.element.ElementActions;
+import com.shaft.tools.io.PropertiesFileManager;
+import com.shaft.tools.io.ReportManager;
+import com.shaft.validation.Assertions;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.ParameterType;
+import io.cucumber.java.Scenario;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+import java.util.Arrays;
+
+public class GuiActionSteps {
+    //TODO: handle shaft_engine listeners
+    private ThreadLocal<WebDriver> driver = new ThreadLocal<>();
+
+    @Given("I Open the target browser")
+    public void BrowserFactory_getBrowser(){
+        driver.set(BrowserFactory.getBrowser());
+    }
+
+    @When("I Navigate to {string}")
+    public void BrowserActions_navigateToURL(String url){
+        BrowserActions.navigateToURL(driver.get(), url);
+    }
+
+    @When("I Navigate to {string} and get redirected to {string}")
+    public void BrowserActions_navigateToURL(String url, String urlAfetrRedirection){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The instance method name 'BrowserActions_navigateToURL' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5707958)